plugins/enthusiast-common/enthusiast_common/builder/validator.py

```diff
@@ -177,28 +177,10 @@ def _validate_retrievers_config(self, retrievers: RetrieversConfig, result: Conf
             result.add_error("retrievers.product configuration is required")
         elif not retrievers.product.retriever_class:
             result.add_error("retrievers.product.retriever_class is required")
-        else:
```

Comment:
Were these validations too much?

Reply:
Yes, as those retrievers may come with different configurations I've placed an `extra_kwargs` field which can handle all cases.
Also we've already spoke about it, this validator needs to be rewrited to Pydatic or something similar. Or probably config itself may be represented as Pydantic classes